PCF-ADR-0003-codeflare-release-process.md

```diff
@@ -0,0 +1,117 @@
+Normal semantic version tags – These will be created whenever components require a new image to be published. This may happen more than once per sprint.
+
+### Testing
+
+Each component will be required to implement unit and e2e tests that are run on all PRs.
+The CodeFlare operator will also include an integration test suite that will ensure stability of the overall stack.
+In addition to running the integration test suite on every PR, we will run it nightly in the CodeFlare operator repository with the `fast` tag for all underlying components
+We are not planning to include integration tests in all component repositories to start.
+
+The components themselves have dependencies on each other as well that may result in updates being required for multiple components. For example, an update to the AppWrapper CRD in the MCAD repository will require us to update all the components below it in the chain. On the other hand, the codeflare-sdk will not accept changes that will break integration with MCAD.
+
+With this in mind, we will make sure that the e2e tests for components pull in the `fast` tag for all components above them in the chain. This will help us catch issues earlier in the sprint.
+
+### Release Process
+
+1. Ask components to generate a new release.
+    1. This will just be the new semantic version tag.
+2. Update image references in CodeFlare Operator repository
+    1. Make sure that any CRD updates in the MCAD repository are copied into the CodeFlare operator repository
+    2. Once merged, create a new operator image.
+3. Generate new operator bundle in CodeFlare Operator
+    1. Update the image reference for the bundle
+    2. As part of this, we will be testing the latest component versions
+    3. Update support matrix in operator README.
+4. Create tag in operator repository and update release notes of operator version to include new support matrix.
+5. Open a pull request to OpenShift community operators repository with latest bundle.
+6. Once merged, update component stable tags to point at the latest image release.
+7. Announce the new release in slack and mail lists, if any.
+8. Update the Distributed Workloads component in ODH.
+    1. At a minimum, we will need to update the README for Distributed Workloads to refer to the latest versions of each component
+    2. Updates could include custom resources, KubeRay version, or usage/installation instructions.
+    3. Update needs to occur within a week of CodeFlare release
+    4. Make sure that the platform SIG is aware of these updates, so that the ODH release notes are updated as expected
+
+Hotfix Process?
+If a critical issues is discovered requiring an out-of-band release, we will follow the normal release process, bumping up the z stream version for the CodeFlare stack as well as the related component.
+
+## Open Questions
+
+1. How does versioning and releases for the distributed workloads component for ODH fit into this?
```

Comment:
In my opinion, every time we cut a new stable release, we should update the ODH to pull in this new version. Ideally we automate that process.

Reply:
Does step 8 of the release process address your point? Or should I add a point in here reiterating that CodeFlare releases will not be considered complete until the distributed workloads component is updated.

Reply:
Ah, yes, I think this is covered by step 8.